Formula/blocky.rb

```diff
@@ -0,0 +1,76 @@
+class Blocky < Formula
+  desc "Fast and lightweight DNS proxy as ad-blocker with many features"
+  homepage "https://0xerr0r.github.io/blocky"
+  url "https://github.com/0xerr0r/blocky/archive/refs/tags/v0.20.tar.gz"
+  sha256 "aae5346e9c1ce4b326b9e578939aa26ddca39338d79d0ddb3eb079ae7a949e87"
+  license "Apache-2.0"
+  head "https://github.com/0xerr0r/blocky.git", branch: "development"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/0xERR0R/blocky/util.Version=#{version}
+      -X github.com/0xERR0R/blocky/util.BuildTime=#{time.iso8601}
+    ]
+    system "go", "build", *std_go_args(ldflags: ldflags)
+    sbin.install_symlink bin/"blocky"
+
+    (var/"log").mkpath
+
+    config_yml = if build.head?
+      <<~EOF_HEAD
+        # Reference the example config in the docs for all options
+        # https://github.com/0xERR0R/blocky/blob/development/docs/config.yml
+        ports:
+          dns: "127.0.0.1:53,[::1]:53"
+        upstream:
+          default:
+            - 1.1.1.1
+            - 1.0.0.1
+        bootstrapDns:
+          - tcp+udp:1.1.1.1
+          - https://1.1.1.1/dns-query
+        log:
+          level: info
+          format: text
+          timestamp: true
+          privacy: false
+      EOF_HEAD
+    else
+      <<~EOF_STABLE
+        # Reference the example config in the docs for all options
+        # https://github.com/0xERR0R/blocky/blob/v0.20/docs/config.yml
+        port: "127.0.0.1:53,[::1]:53"
+        upstream:
+          default:
+            - 1.1.1.1
+            - 1.0.0.1
+        bootstrapDns: tcp+udp:1.1.1.1
+        logLevel: info
+        logFormat: text
+        logTimestamp: true
+        logPrivacy: false
+      EOF_STABLE
+    end
```

Comment:
If upstream doesn't ship one, homebrew shouldn't either

Reply:
The issue is that the service listens on `0.0.0.0` by default. What do you recommend?

Reply:
I recommend upstream ship a config file or change the default.

Reply:
yeah, let's PR that config into the upstream, it would benefit the other package managers as well. :)

Reply:
I submitted that request to the blocky project. https://github.com/0xERR0R/blocky/issues/839

Reply:
We need to wait for a new release with the latest config format changes because some fields have changed in HEAD from the 0.20 release. For instance `port` is now `ports` and `logLevel`/`logFormat`/`logTimestamp` are now in the `logs` hash.

Reply:
Sounds good, thanks!! 👍

Reply:
Doesn't macOS ask users if they want to allow the service to accept incoming network connections? Maybe running blocky without a default config is acceptable since the user decides if they want to allow external access.